exporter/exporter_test.go

```diff
@@ -2244,3 +2244,80 @@ func TestProcessSentinelSlaves(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPScrapeWithPasswordFile(t *testing.T) {
+	if os.Getenv("TEST_REDIS_PWD_FILE") == "" {
+		t.Skipf("TEST_REDIS_PWD_FILE not set - skipping")
+	}
+
+	passwordMap, err := LoadPwdFile(os.Getenv("TEST_REDIS_PWD_FILE"))
+	if err != nil {
+		t.Fatalf("Test Failed, error: %v", err)
+	}
+
+	if len(passwordMap) == 0 {
+		t.Skipf("Password map is empty -skipping")
+	}
+	for target, pwd := range passwordMap {
```

Comment:
Don't iterate over the map that you load but iterate of a given set of tests where you define the inputs and outputs.
For instance, spell out each of the test cases with one `for` loop instead of using `passwordMap` and the double `for` loop`.

See here for more information and examples: https://github.com/golang/go/wiki/TableDrivenTests

Reply:
In particular, this should fail in the CI/CD runs as the IP addresses are invalid.

See [here](https://github.com/oliver006/redis_exporter/blob/master/exporter/exporter_test.go#L1073) for an example how to include target as part of the test and pull it from the env variables.